### DISCUSSION TITLE
How to run tests as part of a monitoring service

### DISCUSSION THREAD

**Discussion by dnmvisser** [internetstandards/Internet.nl#1203] [General]
Hi

I am interested to periodically run tests, as part of a monitoring setup (nagios, in this case). This would allow for notifications if any of the scores change.
For many monitoring tools the basic building block is a script that accepts some parameters, and then returns some status. For this toolset for example it could look like:

```shell
./check_internetnl.sh --domain foobar.nl -test=web --warn 60 --critical 80
```

Or:

```shell
./check_internetnl.sh --domain foobar.nl --test=mail --warn 40 --critical=60
```

I was triggered by the article at https://en.internet.nl/article/release-1.8 and hoping that there would be some container that could be run that does this test, but it's more complex than that.

I read through the doc and the thing that comes closest to what I'm looking for is the [Batch functionality](https://github.com/internetstandards/Internet.nl/blob/main/documentation/Batch.md). Once that is there, the script can be a simple wrapper that does an API call.

Are there any other things I should be aware of?

**Comment by mxsasha** [comment]
We don't have anything for this ourselves, but batch functionality is the way to go here indeed. [Deployment instructions here](https://github.com/internetstandards/Internet.nl/blob/main/documentation/Docker-deployment-batch.md) and there's an [OpenAPI spec of the API](https://redocly.github.io/redoc/?url=https://batch.internet.nl/api/batch/openapi.yaml). If you want to make it extra nice, you could include a link to the results in the output, and/or maybe extract from the technical details what has failed so it can be directly shown in the monitoring setup. And let us know if you publish your work, then we can refer others to it.

**Comment by bwbroersma** [comment]
Nice idea!
Probably this can be made now, some things to consider:
- an API instance should be kept online (cannot run as 'one-off'), mainly because of the Routinator, see [Docker Getting Started](https://github.com/internetstandards/Internet.nl/blob/main/documentation/Docker-getting-started.md#development-environment):
    > RPKI tests rely on Routinator syncing up from external databases and will take a while (`docker logs internetnl-routinator-1 -f`).
- the API is async, so it will need some polling to wait for the result
